src/GdbCommand.h

```diff
@@ -27,28 +28,146 @@ class GdbCommand {
   virtual std::string invoke(GdbServer& gdb_server, Task* t,
                              const std::vector<std::string>& args) = 0;
 
+  /**
+   * When call, gdb will automatically fill argument to this command
+   * by running gdb.execute(). This is useful to gdb state along side
+   * of the command invocation.
```

Comment:
Those are good points. However, I still get the feeling it's possible to simplify the setup. I might have to tinker with it myself.